event-sauce/src/lib.rs

```diff
@@ -69,6 +69,61 @@ pub trait EventData: Serialize + Sized {
 /// Event payloads that can be different variants of an enum.
 pub trait EnumEventData: EventData {}
 
+/// The `EventData` of the `Event` that can be conflicted with an already applied `Event<EDA>`.
+pub trait ConflictCheck<EDA>: EventData
+where
+    EDA: EventData,
+{
+    /// Check if applying `self` is in conflict with an already `applied_event`.
+    ///
+    /// Returns either `Ok(EventData)` in case of no conflicts or [`ConflictData`] with `Self` in
+    /// place of `EDC`, describing the conflict otherwise.
+    ///
+    /// This function will be called during [`UpdateEntityBuilder::try_update`](trait.UpdateEntityBuilder.html#method.try_update).
+    fn conflict_check(self, applied_event: &Event<EDA>) -> Result<Self, ConflictData<EDA, Self>>;
```

Comment:
This method is performing the act of checking for a conflict, so makes more sense to call it `check_conflict`:

```suggestion
    fn check_conflict(self, applied_event: &Event<EDA>) -> Result<Self, ConflictData<EDA, Self>>;
```